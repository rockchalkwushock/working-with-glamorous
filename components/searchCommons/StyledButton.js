import glamorous from 'glamorous'

import { colors } from '../../lib/styles/global'

const StyledButton = glamorous.button(
  {
    border: 'none',
    background: `${colors.ternary}`,
    padding: '0px',
    outline: 'none',
    position: 'relative',
    zIndex: 2, // again must be integer NOT string!
    float: 'right',
    cursor: 'pointer',
    transition: 'all 0.3s ease-in-out'
  },
  ({ open }) => ({
    width: open ? '50px' : '70px',
    height: open ? '50px' : '70px',
    margin: open ? '10px' : null,
    borderRadius: open ? '30px' : '6px'
  })
)

StyledButton.displayName = 'StyledButton'

export default StyledButton
