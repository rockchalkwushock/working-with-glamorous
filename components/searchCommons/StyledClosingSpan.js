import glamorous from 'glamorous'

import { colors } from '../../lib/styles/global'

const StyledClosingSpan = glamorous.span(
  {
    position: 'absolute',
    zIndex: 1,
    top: '24px',
    width: '25px',
    height: '25px',
    cursor: 'pointer',
    '::before': {
      width: '5px',
      height: '25px',
      left: '10px',
      top: '0px',
      position: 'absolute',
      content: '""',
      backgroundColor: `${colors.ternary}`,
      borderRadius: '2px'
    },
    '::after': {
      width: '25px',
      height: '5px',
      left: '0px',
      top: '10px',
      position: 'absolute',
      content: '""',
      backgroundColor: `${colors.ternary}`,
      borderRadius: '2px'
    }
  },
  ({ open }) => ({
    right: open ? '-50px' : '20px',
    transform: open ? 'rotate(45deg)' : 'rotate(-180deg)',
    transition: open
      ? 'all 0.6s cubic-bezier(0.000, 0.105, 0.035, 1.570)'
      : 'all 0.3s cubic-bezier(0.285, -0.450, 0.935, 0.110)',
    transitionDelay: open ? '0.5s' : '0.2s'
  })
)

StyledClosingSpan.displayName = 'StyledClosingSpan'

export default StyledClosingSpan
