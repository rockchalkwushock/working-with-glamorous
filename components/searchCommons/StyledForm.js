import glamorous from 'glamorous'

const StyledForm = glamorous.form({
  display: 'block',
  marginTop: '0em'
})

StyledForm.displayName = 'StyledForm'

const Form = props => <StyledForm {...props} />

export { StyledForm }
export default Form
