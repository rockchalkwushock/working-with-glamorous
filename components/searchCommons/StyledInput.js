import glamorous from 'glamorous'

import { colors, fonts } from '../../lib/styles/global'

const StyledInput = glamorous.input(
  {
    display: 'inline-block',
    width: '100%',
    height: '50px',
    padding: '0px 70px 0px 20px',
    position: 'absolute',
    top: '0',
    left: '0',
    background: 'transparent',
    boxSizing: 'border-box',
    border: 'none',
    outline: 'none',
    fontFamily: `${fonts.primary}`,
    fontSize: '16px',
    fontWeight: 400,
    lineHeight: '20px',
    color: `${colors.emphasis}`,
    transition: 'all 0.3s cubic-bezier(0.000, 0.105, 0.035, 1.570)',
    transitionDelay: '0.3s'
  },
  ({ open }) => ({
    opacity: open ? 1 : 0,
    transform: open ? 'translate(0, 10px)' : 'translate(0, 60px)'
  })
)

StyledInput.displayName = 'StyledInput'

const Input = props => <StyledInput {...props} {...props.input} />

export { StyledInput }
export default Input
