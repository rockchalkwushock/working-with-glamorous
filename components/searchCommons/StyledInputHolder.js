import glamorous from 'glamorous'

const StyledInputHolder = glamorous.div(
  {
    display: 'block',
    overflow: 'hidden',
    height: '70px',
    position: 'relative'
  },
  ({ open }) => ({
    width: open ? '450px' : '70px',
    borderRadius: open ? '50px' : '6px',
    backgroundColor: open === true ? 'rgba(0,0,0,0.5)' : 'rgba(255,255,255,0)',
    transition: open === true
      ? 'all 0.5s cubic-bezier(0.000, 0.105, 0.035, 1.570)'
      : 'all 0.3s ease-in-out'
  })
)

StyledInputHolder.displayName = 'StyledInputHolder'

export default StyledInputHolder
