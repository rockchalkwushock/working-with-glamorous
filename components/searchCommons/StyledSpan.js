import glamorous from 'glamorous'

import { colors } from '../../lib/styles/global'

const StyledSpan = glamorous.span(
  {
    width: '22px',
    height: '22px',
    display: 'inline-block',
    verticalAlign: 'middle',
    position: 'relative',
    transition: 'all 0.4s cubic-bezier(0.650, -0.600, 0.240, 1.650)',
    '::before': {
      width: '4px',
      height: '11px',
      left: '9px',
      top: '18px',
      borderRadius: '2px',
      backgroundColor: `${colors.emphasis}`,
      position: 'absolute',
      content: '""'
    },
    '::after': {
      width: '14px',
      height: '14px',
      left: '0px',
      top: '0px',
      borderRadius: '16px',
      border: `4px solid ${colors.emphasis}`,
      position: 'absolute',
      content: '""'
    }
  },
  ({ open }) => ({
    transform: open ? 'rotate(-45deg)' : 'rotate(45deg)'
  })
)

StyledSpan.displayName = 'StyledSpan'

export default StyledSpan
