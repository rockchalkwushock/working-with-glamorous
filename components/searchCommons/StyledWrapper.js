import glamorous from 'glamorous'

const StyledWrapper = glamorous.div({
  display: 'block',
  position: 'absolute',
  transform: 'translate(-50%, -50%)',
  top: '40%',
  left: '50%'
})

StyledWrapper.displayName = 'StyledWrapper'

export default StyledWrapper
