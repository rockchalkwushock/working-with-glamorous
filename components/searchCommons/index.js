import Form, { StyledForm } from './StyledForm'
import StyledButton from './StyledButton'
import StyledClosingSpan from './StyledClosingSpan'
import Input, { StyledInput } from './StyledInput'
import StyledInputHolder from './StyledInputHolder'
import StyledSpan from './StyledSpan'
import StyledWrapper from './StyledWrapper'

export {
  Form,
  Input,
  StyledButton,
  StyledClosingSpan,
  StyledForm,
  StyledInput,
  StyledInputHolder,
  StyledSpan,
  StyledWrapper
}
