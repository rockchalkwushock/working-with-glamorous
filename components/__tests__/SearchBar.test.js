import { shallow, render, mount } from 'enzyme'

import { SearchBar } from '../index'
import { StyledWrapper } from '../searchCommons'

describe('SearchBar Component', () => {
  let ui
  test('should render to the DOM', () => {
    ui = <SearchBar />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
  // replicates above test, should probably just remove it
  test('should render initial styling when props.open: false', () => {
    ui = <StyledWrapper open={false} />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
  test('should render animated styling when props.open: true', () => {
    ui = <StyledWrapper open />
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
})
