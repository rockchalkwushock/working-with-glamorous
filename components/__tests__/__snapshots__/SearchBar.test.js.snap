// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-1wvnu8l,
[data-css-1wvnu8l] {
  display: block;
  margin-top: 0em;
}

.css-mfx0z8,
[data-css-mfx0z8] {
  -webkit-transform: translate(-50%, -50%);
  display: block;
  position: absolute;
  transform: translate(-50%, -50%);
  top: 40%;
  left: 50%;
}

.css-1ypye1o,
[data-css-1ypye1o] {
  -webkit-transition: all 0.3s ease-in-out;
  display: block;
  overflow: hidden;
  height: 70px;
  position: relative;
  width: 70px;
  border-radius: 6px;
  background-color: rgba(255,255,255,0);
  transition: all 0.3s ease-in-out;
}

.css-1gjdfoc,
[data-css-1gjdfoc] {
  -webkit-transform: translate(0, 60px);
  -moz-box-sizing: border-box;
  -webkit-transition: all 0.3s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  -webkit-transition-delay: 0.3s;
  position: absolute;
  top: 0px;
  left: 0px;
  background: transparent;
  box-sizing: border-box;
  border: none;
  outline: none;
  font-family: "Questrial", sans-serif;
  width: 100%;
  font-weight: 400;
  line-height: 20px;
  color: #FEBA00;
  transition: all 0.3s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  transition-delay: 0.3s;
  opacity: 0;
  transform: translate(0, 60px);
  display: inline-block;
  height: 50px;
  padding: 0px 70px 0px 20px;
  font-size: 16px;
}

.css-17ppabt,
[data-css-17ppabt] {
  -webkit-transition: all 0.3s ease-in-out;
  border: none;
  padding: 0px;
  outline: none;
  position: relative;
  z-index: 2;
  background: #0C98EB;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  width: 70px;
  height: 70px;
  border-radius: 6px;
  float: right;
}

.css-7veum2,
[data-css-7veum2] {
  -webkit-transition: all 0.4s cubic-bezier(0.650, -0.600, 0.240, 1.650);
  -webkit-transform: rotate(45deg);
  width: 22px;
  height: 22px;
  display: inline-block;
  vertical-align: middle;
  position: relative;
  transition: all 0.4s cubic-bezier(0.650, -0.600, 0.240, 1.650);
  transform: rotate(45deg);
}

.css-7veum2::before,
[data-css-7veum2]::before {
  width: 4px;
  height: 11px;
  left: 9px;
  top: 18px;
  border-radius: 2px;
  background-color: #FEBA00;
  position: absolute;
  content: "";
}

.css-7veum2::after,
[data-css-7veum2]::after {
  width: 14px;
  height: 14px;
  left: 0px;
  top: 0px;
  border-radius: 16px;
  border: 4px solid #FEBA00;
  position: absolute;
  content: "";
}

.css-8pl84a,
[data-css-8pl84a] {
  -webkit-transition-delay: 0.2s;
  -webkit-transform: rotate(-180deg);
  -webkit-transition: all 0.3s cubic-bezier(0.285, -0.450, 0.935, 0.110);
  width: 25px;
  height: 25px;
  cursor: pointer;
  z-index: 1;
  transform: rotate(-180deg);
  transition: all 0.3s cubic-bezier(0.285, -0.450, 0.935, 0.110);
  transition-delay: 0.2s;
  position: absolute;
  top: 24px;
  right: 20px;
}

.css-8pl84a::before,
[data-css-8pl84a]::before {
  width: 5px;
  height: 25px;
  left: 10px;
  top: 0px;
  position: absolute;
  content: "";
  background-color: #0C98EB;
  border-radius: 2px;
}

.css-8pl84a::after,
[data-css-8pl84a]::after {
  width: 25px;
  height: 5px;
  left: 0px;
  top: 10px;
  position: absolute;
  content: "";
  background-color: #0C98EB;
  border-radius: 2px;
}

<withState(withState(withHandlers(SearchBar)))>
  <withState(withHandlers(SearchBar))
    toggle={[Function]}
    view={false}
  >
    <withHandlers(SearchBar)
      toggle={[Function]}
      updateValue={[Function]}
      value=""
      view={false}
    >
      <Form
        onSubmit={[Function]}
      >
        <StyledForm
          onSubmit={[Function]}
        >
          <form
            className="css-1wvnu8l"
            onSubmit={[Function]}
          >
            <StyledWrapper>
              <div
                className="css-mfx0z8"
              >
                <StyledInputHolder
                  open={false}
                >
                  <div
                    className="css-1ypye1o"
                  >
                    <Input
                      onChange={[Function]}
                      open={false}
                      placeholder="Search a topic on StackOverflow"
                      type="text"
                      value=""
                    >
                      <StyledInput
                        onChange={[Function]}
                        open={false}
                        placeholder="Search a topic on StackOverflow"
                        type="text"
                        value=""
                      >
                        <input
                          className="css-1gjdfoc"
                          onChange={[Function]}
                          placeholder="Search a topic on StackOverflow"
                          type="text"
                          value=""
                        />
                      </StyledInput>
                    </Input>
                    <StyledButton
                      onClick={[Function]}
                      open={false}
                      type="button"
                    >
                      <button
                        className="css-17ppabt"
                        onClick={[Function]}
                        type="button"
                      >
                        <StyledSpan
                          open={false}
                        >
                          <span
                            className="css-7veum2"
                          />
                        </StyledSpan>
                      </button>
                    </StyledButton>
                  </div>
                </StyledInputHolder>
                <StyledClosingSpan
                  onClick={[Function]}
                  open={false}
                >
                  <span
                    className="css-8pl84a"
                    onClick={[Function]}
                  />
                </StyledClosingSpan>
              </div>
            </StyledWrapper>
          </form>
        </StyledForm>
      </Form>
    </withHandlers(SearchBar)>
  </withState(withHandlers(SearchBar))>
</withState(withState(withHandlers(SearchBar)))>
`;

exports[`enzyme.mount 2`] = `
.css-mfx0z8,
[data-css-mfx0z8] {
  -webkit-transform: translate(-50%, -50%);
  display: block;
  position: absolute;
  transform: translate(-50%, -50%);
  top: 40%;
  left: 50%;
}

<StyledWrapper
  open={true}
>
  <div
    className="css-mfx0z8"
  />
</StyledWrapper>
`;

exports[`enzyme.render 1`] = `
.css-1wvnu8l,
[data-css-1wvnu8l] {
  display: block;
  margin-top: 0em;
}

.css-mfx0z8,
[data-css-mfx0z8] {
  -webkit-transform: translate(-50%, -50%);
  display: block;
  position: absolute;
  transform: translate(-50%, -50%);
  top: 40%;
  left: 50%;
}

.css-1ypye1o,
[data-css-1ypye1o] {
  -webkit-transition: all 0.3s ease-in-out;
  display: block;
  overflow: hidden;
  height: 70px;
  position: relative;
  width: 70px;
  border-radius: 6px;
  background-color: rgba(255,255,255,0);
  transition: all 0.3s ease-in-out;
}

.css-1gjdfoc,
[data-css-1gjdfoc] {
  -webkit-transform: translate(0, 60px);
  -moz-box-sizing: border-box;
  -webkit-transition: all 0.3s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  -webkit-transition-delay: 0.3s;
  position: absolute;
  top: 0px;
  left: 0px;
  background: transparent;
  box-sizing: border-box;
  border: none;
  outline: none;
  font-family: "Questrial", sans-serif;
  width: 100%;
  font-weight: 400;
  line-height: 20px;
  color: #FEBA00;
  transition: all 0.3s cubic-bezier(0.000, 0.105, 0.035, 1.570);
  transition-delay: 0.3s;
  opacity: 0;
  transform: translate(0, 60px);
  display: inline-block;
  height: 50px;
  padding: 0px 70px 0px 20px;
  font-size: 16px;
}

.css-17ppabt,
[data-css-17ppabt] {
  -webkit-transition: all 0.3s ease-in-out;
  border: none;
  padding: 0px;
  outline: none;
  position: relative;
  z-index: 2;
  background: #0C98EB;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  width: 70px;
  height: 70px;
  border-radius: 6px;
  float: right;
}

.css-7veum2,
[data-css-7veum2] {
  -webkit-transition: all 0.4s cubic-bezier(0.650, -0.600, 0.240, 1.650);
  -webkit-transform: rotate(45deg);
  width: 22px;
  height: 22px;
  display: inline-block;
  vertical-align: middle;
  position: relative;
  transition: all 0.4s cubic-bezier(0.650, -0.600, 0.240, 1.650);
  transform: rotate(45deg);
}

.css-7veum2::before,
[data-css-7veum2]::before {
  width: 4px;
  height: 11px;
  left: 9px;
  top: 18px;
  border-radius: 2px;
  background-color: #FEBA00;
  position: absolute;
  content: "";
}

.css-7veum2::after,
[data-css-7veum2]::after {
  width: 14px;
  height: 14px;
  left: 0px;
  top: 0px;
  border-radius: 16px;
  border: 4px solid #FEBA00;
  position: absolute;
  content: "";
}

.css-8pl84a,
[data-css-8pl84a] {
  -webkit-transition-delay: 0.2s;
  -webkit-transform: rotate(-180deg);
  -webkit-transition: all 0.3s cubic-bezier(0.285, -0.450, 0.935, 0.110);
  width: 25px;
  height: 25px;
  cursor: pointer;
  z-index: 1;
  transform: rotate(-180deg);
  transition: all 0.3s cubic-bezier(0.285, -0.450, 0.935, 0.110);
  transition-delay: 0.2s;
  position: absolute;
  top: 24px;
  right: 20px;
}

.css-8pl84a::before,
[data-css-8pl84a]::before {
  width: 5px;
  height: 25px;
  left: 10px;
  top: 0px;
  position: absolute;
  content: "";
  background-color: #0C98EB;
  border-radius: 2px;
}

.css-8pl84a::after,
[data-css-8pl84a]::after {
  width: 25px;
  height: 5px;
  left: 0px;
  top: 10px;
  position: absolute;
  content: "";
  background-color: #0C98EB;
  border-radius: 2px;
}

<form
  class="css-1wvnu8l"
>
  <div
    class="css-mfx0z8"
  >
    <div
      class="css-1ypye1o"
    >
      <input
        class="css-1gjdfoc"
        placeholder="Search a topic on StackOverflow"
        type="text"
        value=""
      />
      <button
        class="css-17ppabt"
        type="button"
      >
        <span
          class="css-7veum2"
        />
      </button>
    </div>
    <span
      class="css-8pl84a"
    />
  </div>
</form>
`;

exports[`enzyme.render 2`] = `
.css-mfx0z8,
[data-css-mfx0z8] {
  -webkit-transform: translate(-50%, -50%);
  display: block;
  position: absolute;
  transform: translate(-50%, -50%);
  top: 40%;
  left: 50%;
}

<div
  class="css-mfx0z8"
/>
`;

exports[`enzyme.shallow 1`] = `
<withState(withHandlers(SearchBar))
  toggle={[Function]}
  view={false}
/>
`;

exports[`enzyme.shallow 2`] = `
.css-mfx0z8,
[data-css-mfx0z8] {
  -webkit-transform: translate(-50%, -50%);
  display: block;
  position: absolute;
  transform: translate(-50%, -50%);
  top: 40%;
  left: 50%;
}

<div
  className="css-mfx0z8"
/>
`;
